### DISCUSSION TITLE
关于上下文采集设计的疑问

### DISCUSSION THREAD

**Discussion by XLPE** [alibaba/ilogtail#992] [Ideas]
看到社区分享过这篇文章[使用日志上下文聚合插件使能上下文查询及Livetail](https://developer.aliyun.com/article/1066553)，请问一下当初设计的时候，为啥会用一个组的设计，没固定一个组，然后自增ID持久化下来，保证自增ID总是连续呢？
现在的设计有一个问题，因为组ID和组内偏移都是保存在当前内存，重启后，无法查看上下文了。

**Comment by messixukejia** [comment]
这个是配合sls服务端的功能，不具备普适性。

**Comment by yyuuttaaoo** [comment]
此外，这个也是有意为之，因为如果ilogtail崩溃，日志会从上个检查点开始采集，这部分可能重复，如果继续使用原来的id，那么一条日志就可能会有多个上下文，更加不好处理。就目前使用情况来看，这并没有成为实际使用中的问题，如果觉得有更好的方式，可以留言。
